### PR TITLE
[opencv/en]

### PR DESCRIPTION
**Pull request by yogeshojha**
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] I solemnly swear that this is all original content of which I am the original author
- [x] Pull request title is prepended with `[language/lang-code]`
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- [x] If you've changed any part of the YAML Frontmatter, make sure it is formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [x] Yes, I have double-checked quotes and field names!

### CODE REVIEW

File: opencv.html.markdown [hunk 1]
```diff
@@ -0,0 +1,155 @@
+---
+language: python
```
Comment (divayprakash) [inline opencv.html.markdown]: Please change this to

```
tool: OpenCV
```

File: opencv.html.markdown [hunk 2]
```diff
@@ -0,0 +1,155 @@
+---
```
Comment (divayprakash) [inline opencv.html.markdown]: Add this line here

```
category: tool
```

**Comment by divayprakash** [comment]
@yogeshojha Thank you for the contribution and welcome to the LXIYM community! Please make the changes as above and I'll merge this in ASAP!

File: opencv.html.markdown [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+<https://opencv.org/>
+
+Additional resources may be found at 
+<https://en.wikipedia.org/wiki/OpenCV>
+
+Good OpenCv Tutorials
+<https://opencv-python-tutroals.readthedocs.io/en/latest/py_tutorials/py_tutorials.html>
+<https://realpython.com/python-opencv-color-spaces/>
+<https://pyimagesearch.com>
+<https://www.learnopencv.com>
```
Comment (divayprakash) [inline opencv.html.markdown]: The whole section of 'Further Reading' has to be formatted as list along with correct link formatting as follows -

```
### Further Reading:

* Download Cascade from [https://github.com/opencv/opencv/blob/master/data/haarcascades/]()
* OpenCV drawing Functions [https://docs.opencv.org/2.4/modules/core/doc/drawing_functions.html]()
* An up-to-date language reference can be found at [https://opencv.org/]()
* Additional resources may be found at [https://en.wikipedia.org/wiki/OpenCV]()
* Good OpenCV Tutorials
	* [https://opencv-python-tutroals.readthedocs.io/en/latest/py_tutorials/py_tutorials.html]()
	* [https://realpython.com/python-opencv-color-spaces/]()
	* [https://pyimagesearch.com]()
	* [https://www.learnopencv.com]()
```

File: opencv.html.markdown [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+* Mac Installation Instructions (High Sierra):
+https://medium.com/@nuwanprabhath/installing-opencv-in-macos-high-sierra-for-python-3-89c79f0a246a
+
+* Linux Installation Instructions (Ubuntu 18.04):
+<https://www.pyimagesearch.com/2018/05/28/ubuntu-18-04-how-to-install-opencv/>
+
+### Here we will be focusing on python implementation of OpenCV
+
+```
```
Comment (divayprakash) [inline opencv.html.markdown]: This needs to be changed to 

```
```python
```

for the file to be formatted correctly.

**Comment by divayprakash** [comment]
Please make the changes as requested.

File: opencv.html.markdown [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+---
+### Opencv
+
+OpenCV (Open Source Computer Vision) is a library of programming functions mainly aimed at real-time computer vision. 
+Originally developed by Intel, it was later supported by Willow Garage then Itseez (which was later acquired by Intel). 
+Opencv currently supports wide variety of languages like, C++, Python, Java etc
+
+#### Installation
+Please refer to these articles for installation of OpenCV on your computer.
+* Windows Installation Instructions: [https://opencv-python-tutroals.readthedocs.io/en/latest/py_tutorials/py_setup/py_setup_in_windows/py_setup_in_windows.html#install-opencv-python-in-windows]
```
Comment (divayprakash) [inline opencv.html.markdown]: Please add a newline before this line ie.

```
Please refer to these articles for installation of OpenCV on your computer.

* Windows Installation Instructions
```

Comment (divayprakash) [reply opencv.html.markdown]: Also, insert a `()` at the end of the link ie.

```
* Windows Installation Instructions: [https://opencv-python-tutroals.readthedocs.io/en/latest/py_tutorials/py_setup/py_setup_in_windows/py_setup_in_windows.html#install-opencv-python-in-windows]()
```

This has to be done for **all** links present in this list.

File: opencv.html.markdown [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+* Mac Installation Instructions (High Sierra): [https://medium.com/@nuwanprabhath/installing-opencv-in-macos-high-sierra-for-python-3-89c79f0a246a]
+* Linux Installation Instructions (Ubuntu 18.04): [https://www.pyimagesearch.com/2018/05/28/ubuntu-18-04-how-to-install-opencv]
+### Here we will be focusing on python implementation of OpenCV
```
Comment (divayprakash) [inline opencv.html.markdown]: Insert a newline before this line ie.

```
* Linux Installation Instructions (Ubuntu 18.04): [https://www.pyimagesearch.com/2018/05/28/ubuntu-18-04-how-to-install-opencv]()

### Here we will be focusing on python implementation of OpenCV
```

File: opencv.html.markdown [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+cv2.waitKey(0)
+
+cv2.destroyAllWindows()
+# destroyAllWindows() destroys all windows. 
+# If you wish to destroy specific window pass the exact name of window you created.
+```
+
+### Further Reading:
+
+* Download Cascade from [https://github.com/opencv/opencv/blob/master/data/haarcascades]
```
Comment (divayprakash) [inline opencv.html.markdown]: Every link in the list has to have a `()` at the end ie.

```
* Download Cascade from [https://github.com/opencv/opencv/blob/master/data/haarcascades]()
```

**Comment by yogeshojha** [comment]
Made necessary Changes

File: opencv.html.markdown (hunk 6 above)
Comment (divayprakash) [reply opencv.html.markdown]: @yogeshojha This one is still left. Please fix this as it is not rendering correctly.

**Comment by divayprakash** [comment]
@yogeshojha https://github.com/adambard/learnxinyminutes-docs/pull/3259#discussion_r224962651 Please check this one out.

**Comment by divayprakash** [comment]
@yogeshojha There needs to be one empty line before and after this statement -
```
## Here we will be focusing on python implementation of OpenCV
```

**Comment by yogeshojha** [comment]
Sorry, I missed it. I have now inserted new line before and after 

```## Here we will be focusing on python implementation of OpenCV```

**Comment by divayprakash** [comment]
@yogeshojha Thank you for your contribution. This has been a great review exercise and you really worked on the PR. Welcome to the LXIYM community!

**Comment by yogeshojha** [comment]
Thank you so much @divayprakash !

**Pull request merged**